core/src/main/java/hudson/model/User.java

```diff
@@ -713,8 +714,9 @@ public static boolean isIdOrFullnameAllowed(@CheckForNull String id) {
         if (id == null || StringUtils.isBlank(id)) {
             return false;
         }
+        final String trimmedId = id.trim();
```

Comment:
I would rather name it something like `normalizedId` and move that normalization to a dedicated method. That method would only contain the current `trim()` currently but might be improved progressively.